### DISCUSSION TITLE
How do I avoid cascade issues within Nuxt 2 components?

### DISCUSSION THREAD

**Discussion by andrewCodes** [nuxt/nuxt#26659] [Questions (Nuxt 2)]
I'm experiencing styling issues in my Nuxt app caused by incorrect ordering of styles. I was previously writing my CSS (SCSS) in dedicated .scss files so I was manually able to order styles to make use of the cascade. Take the following components as an example of the issue (I've removed code for brevity):

```
<!-- asset.vue -->

<template>
    <div class="asset">
        <!-- rest of code -->
    </div>
</template>

<!-- asset-content.vue -->

<template>
    <section class="asset-content">
        <div class="asset-content__container">
            <asset class="asset-content__asset"/>
            <content-area class="asset-content__content"/>
        </div>
    </section>
</template>
```

As you can see, the `<asset/>` component is used inside the `<asset-content/>` component. Both `.asset` and `.asset-content__asset` have the same specificity. When storing my SCSS in .scss files the specificity was not an issue as I used the cascade to ensure any styling on `.asset-content__asset` wrote over styling on `.asset`. For example:

```
.asset {
    position: relative;
}
.asset-content__asset {
    position: absolute;
}
```

However now I have moved my SCSS into the vue files I have lost control over the cascade and so, in places, `.asset` is seemingly being loaded in after `.asset-content__asset` and so `.asset` is over-riding `.asset-content__asset`.

How do I overcome this issue and ensure styles are imported in the order I want them to be. I appreciate there are other solutions like using `scoped` on the `<style>` in the vue files or increasing the specificity of `.asset-content__asset` but neither seem to be the correct solution. Using `scoped` is going to result in more CSS than I need as each time I use the `<asset-content/>` module a new set of CSS is going to be added. Increasing the specificity of `.asset-content__asset` is simply a workaround that, whilst solving the problem, does not directly address it.

In short, is there any way in Nuxt to dictate the order of classes when your CSS / SCSS is in vue files like you can when your CSS / SCSS is in scss files?
